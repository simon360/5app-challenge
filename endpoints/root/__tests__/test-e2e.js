const root = require("../");

describe("End to end tests", () => {
  it("should match sample request to response", () => {
    const request = require("./samples/request.json");
    const response = require("./samples/response.json");

    expect(root(request)).toMatchObject(response);
  });

  it("should match error snapshot", () => {
    expect(
      root({
        payload: [{}]
      })
    ).toMatchSnapshot();
  });
});
