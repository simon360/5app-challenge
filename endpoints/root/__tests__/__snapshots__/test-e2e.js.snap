// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`End to end tests should match error snapshot 1`] = `
Object {
  "errors": Array [
    Object {
      "dataPath": ".payload[0]",
      "keyword": "required",
      "message": "should have required property 'count'",
      "params": Object {
        "missingProperty": "count",
      },
      "schemaPath": "#/properties/payload/items/required",
    },
  ],
}
`;
